### PR TITLE
Use details HTML for usage message

### PR DESCRIPTION
**Pull request by TravisEz13**

### CODE REVIEW

File: FunctionApp/Modules/PoshChan/PoshChan.psm1 [hunk 1]
```diff
@@ -39,22 +39,81 @@ function Test-User($user, $settings, $setting) {
...
+}
+
+# Convert VERY simple markdown with codeblocks to HTML
+function Convert-CodeMarkdownToHTML
+{
+    param(
+        $markdown
+    )
+
+    if($markdown -notmatch '`')
```
Comment (SteveL-MSFT) [inline FunctionApp/Modules/PoshChan/PoshChan.psm1]:
```suggestion
    if ($markdown -notmatch '`')
```

File: FunctionApp/Modules/PoshChan/PoshChan.psm1 [hunk 2]
```diff
@@ -39,22 +39,81 @@ function Test-User($user, $settings, $setting) {
...
+    {
+        return $markdown
+    }
+
+    $markdownParts = $markdown -split '`'
+
+    if($markdownParts.Count % 2 -ne 1)
```
Comment (SteveL-MSFT) [inline FunctionApp/Modules/PoshChan/PoshChan.psm1]:
```suggestion
    if ($markdownParts.Count % 2 -ne 1)
```

File: FunctionApp/Modules/PoshChan/PoshChan.psm1 [hunk 3]
```diff
@@ -39,22 +39,81 @@ function Test-User($user, $settings, $setting) {
...
+    {
+        throw 'Invalid formed markdown'
+    }
+
+    $sb = [System.Text.StringBuilder]::new()
+
+    $max = ($markdownParts.Count -1)
+    0..($markdownParts.Count -1)| ForEach-Object {
```
Comment (SteveL-MSFT) [inline FunctionApp/Modules/PoshChan/PoshChan.psm1]:
```suggestion
    0..($markdownParts.Count -1) | ForEach-Object {
```

File: FunctionApp/Modules/PoshChan/PoshChan.psm1 [hunk 4]
```diff
@@ -39,22 +39,81 @@ function Test-User($user, $settings, $setting) {
...
+        $part = $_
+        if($part -eq 0)
```
Comment (SteveL-MSFT) [inline FunctionApp/Modules/PoshChan/PoshChan.psm1]:
```suggestion
        if ($part -eq 0)
```

File: FunctionApp/Modules/PoshChan/PoshChan.psm1 [hunk 5]
```diff
@@ -39,22 +39,81 @@ function Test-User($user, $settings, $setting) {
...
+        {
+            $null=$sb.Append($markdownParts[$part])
```
Comment (SteveL-MSFT) [inline FunctionApp/Modules/PoshChan/PoshChan.psm1]:
```suggestion
            $null = $sb.Append($markdownParts[$part])
```

File: FunctionApp/Modules/PoshChan/PoshChan.psm1 [hunk 6]
```diff
@@ -39,22 +39,81 @@ function Test-User($user, $settings, $setting) {
...
+        }
+        else
+        {
+            switch($_ % 2)
+            {
+                1 {
+                    $null=$sb.Append('<code>')
```
Comment (SteveL-MSFT) [inline FunctionApp/Modules/PoshChan/PoshChan.psm1]:
```suggestion
                    $null = $sb.Append('<code>')
```

File: FunctionApp/Modules/PoshChan/PoshChan.psm1 [hunk 7]
```diff
@@ -39,22 +39,81 @@ function Test-User($user, $settings, $setting) {
...
+    $max = ($markdownParts.Count -1)
+    0..($markdownParts.Count -1)| ForEach-Object {
+        $part = $_
+        if($part -eq 0)
+        {
+            $null=$sb.Append($markdownParts[$part])
+        }
+        else
+        {
+            switch($_ % 2)
```
Comment (SteveL-MSFT) [inline FunctionApp/Modules/PoshChan/PoshChan.psm1]:
```suggestion
            switch ($_ % 2)
```

File: FunctionApp/Modules/PoshChan/PoshChan.psm1 [hunk 8]
```diff
@@ -39,22 +39,81 @@ function Test-User($user, $settings, $setting) {
...
+            {
+                1 {
+                    $null=$sb.Append('<code>')
+                    $null=$sb.Append($markdownParts[$part])
```
Comment (SteveL-MSFT) [inline FunctionApp/Modules/PoshChan/PoshChan.psm1]:
```suggestion
                    $null = $sb.Append($markdownParts[$part])
```

File: FunctionApp/Modules/PoshChan/PoshChan.psm1 [hunk 9]
```diff
@@ -39,22 +39,81 @@ function Test-User($user, $settings, $setting) {
...
+                }
+                0 {
+                    $null=$sb.Append('</code>')
```
Comment (SteveL-MSFT) [inline FunctionApp/Modules/PoshChan/PoshChan.psm1]:
```suggestion
                    $null = $sb.Append('</code>')
```

File: FunctionApp/Modules/PoshChan/PoshChan.psm1 [hunk 10]
```diff
@@ -39,22 +39,81 @@ function Test-User($user, $settings, $setting) {
...
+                    $null=$sb.Append($markdownParts[$part])
```
Comment (SteveL-MSFT) [inline FunctionApp/Modules/PoshChan/PoshChan.psm1]:
```suggestion
                    $null = $sb.Append($markdownParts[$part])
```

**Comment by SteveL-MSFT** [comment]
@PoshChan-Staging help

**Comment by PoshChan-Staging** [comment]
@SteveL-MSFT, all requests start with the magic word: `Please`

<details>
<summary>Commands available in this repo for you:
</summary>
<ul><li>
<code>retry &lt;target&gt;</code> this will attempt to retry only the failed jobs for the target pipeline, <code>restart</code> can be used in place of <code>retry</code>
</li>
<li>
<code>rebuild &lt;target&gt;</code> this will perform a complete rebuild of the target pipeline, <code>rerun</code> can be used in place of <code>rebuild</code> Supported values for &lt;target&gt; which can be a comma separated list are: <code>ci</code>
</li>
<li>
<code>remind me in &lt;value&gt; &lt;units&gt;</code> this will create a reminder that will be posted after the specified duration &lt;value&gt; is a number, and &lt;units&gt; can be <code>minutes</code>, <code>hours</code>, or <code>days</code> (singular or plural)
</li>

</ul>
</details>

**Comment by SteveL-MSFT** [comment]
@PoshChan help

**Comment by PoshChan** [comment]
@SteveL-MSFT, all requests start with the magic word: `Please`

<details>
<summary>Commands available in this repo for you:
</summary>
<ul><li>
<code>retry &lt;target&gt;</code> this will attempt to retry only the failed jobs for the target pipeline, <code>restart</code> can be used in place of <code>retry</code>
</li>
<li>
<code>rebuild &lt;target&gt;</code> this will perform a complete rebuild of the target pipeline, <code>rerun</code> can be used in place of <code>rebuild</code> Supported values for &lt;target&gt; which can be a comma separated list are: <code>ci</code>
</li>
<li>
<code>remind me in &lt;value&gt; &lt;units&gt;</code> this will create a reminder that will be posted after the specified duration &lt;value&gt; is a number, and &lt;units&gt; can be <code>minutes</code>, <code>hours</code>, or <code>days</code> (singular or plural)
</li>

</ul>
</details>

**Pull request merged**